var/spack/repos/builtin/packages/libmad/libmad-0.15.1b.patch

```diff
@@ -0,0 +1,108 @@
+@@ -0,0 +1 @@
++# patch
+diff -Naur libmad-0.15.1b.orig/ChangeLog libmad-0.15.1b/ChangeLog
+--- libmad-0.15.1b.orig/ChangeLog	1970-01-01 02:00:00.000000000 +0200
++++ libmad-0.15.1b/ChangeLog	2019-07-09 16:12:02.871091228 +0300
+@@ -0,0 +1 @@
++# patch
+diff -Naur libmad-0.15.1b.orig/configure.ac libmad-0.15.1b/configure.ac
+--- libmad-0.15.1b.orig/configure.ac	2019-07-09 16:03:27.560110809 +0300
++++ libmad-0.15.1b/configure.ac	2019-07-09 16:04:03.960038510 +0300
+@@ -28,7 +28,7 @@
+ 
+ AM_INIT_AUTOMAKE
+ 
+-AM_CONFIG_HEADER([config.h])
++AC_CONFIG_HEADERS([config.h])
+ 
+ dnl System type.
+ 
+@@ -124,71 +124,7 @@
+ 
+ if test "$GCC" = yes
+ then
+-    if test -z "$arch"
+-    then
+-	case "$host" in
+-	    i386-*)           ;;
+-	    i?86-*)           arch="-march=i486" ;;
+-	    arm*-empeg-*)     arch="-march=armv4 -mtune=strongarm1100" ;;
+-	    armv4*-*)         arch="-march=armv4 -mtune=strongarm" ;;
+-	    powerpc-*)        ;;
+-	    mips*-agenda-*)   arch="-mcpu=vr4100" ;;
+-	    mips*-luxsonor-*) arch="-mips1 -mcpu=r3000 -Wa,-m4010" ;;
+-	esac
+-    fi
+-
+-    case "$optimize" in
+-	-O|"-O "*)
+-	    optimize="-O"
+-	    optimize="$optimize -fforce-mem"
+-	    optimize="$optimize -fforce-addr"
+-	    : #x optimize="$optimize -finline-functions"
+-	    : #- optimize="$optimize -fstrength-reduce"
+-	    optimize="$optimize -fthread-jumps"
+-	    optimize="$optimize -fcse-follow-jumps"
+-	    optimize="$optimize -fcse-skip-blocks"
+-	    : #x optimize="$optimize -frerun-cse-after-loop"
+-	    : #x optimize="$optimize -frerun-loop-opt"
+-	    : #x optimize="$optimize -fgcse"
+-	    optimize="$optimize -fexpensive-optimizations"
+-	    optimize="$optimize -fregmove"
+-	    : #* optimize="$optimize -fdelayed-branch"
+-	    : #x optimize="$optimize -fschedule-insns"
+-	    optimize="$optimize -fschedule-insns2"
+-	    : #? optimize="$optimize -ffunction-sections"
+-	    : #? optimize="$optimize -fcaller-saves"
+-	    : #> optimize="$optimize -funroll-loops"
+-	    : #> optimize="$optimize -funroll-all-loops"
+-	    : #x optimize="$optimize -fmove-all-movables"
+-	    : #x optimize="$optimize -freduce-all-givs"
+-	    : #? optimize="$optimize -fstrict-aliasing"
+-	    : #* optimize="$optimize -fstructure-noalias"
+-
+-	    case "$host" in
+-		arm*-*)
+-		    optimize="$optimize -fstrength-reduce"
+-		    ;;
+-		mips*-*)
+-		    optimize="$optimize -fstrength-reduce"
+-		    optimize="$optimize -finline-functions"
+-		    ;;
+-		i?86-*)
+-		    optimize="$optimize -fstrength-reduce"
+-		    ;;
+-		powerpc-apple-*)
+-		    # this triggers an internal compiler error with gcc2
+-		    : #optimize="$optimize -fstrength-reduce"
+-
+-		    # this is really only beneficial with gcc3
+-		    : #optimize="$optimize -finline-functions"
+-		    ;;
+-		*)
+-		    # this sometimes provokes bugs in gcc 2.95.2
+-		    : #optimize="$optimize -fstrength-reduce"
+-		    ;;
+-	    esac
+-	    ;;
+-    esac
++    optimize="-O2"
```

Comment:
I'm a little worried about this patch. Removing all optimization flags and always using `-O2` sounds like overkill to me. Where does this patch come from? Has it been submitted upstream to the developers?

Reply:
Hi,

I should have added some mentions about the patch source. The patch is from [here](http://www.linuxfromscratch.org/blfs/view/svn/multimedia/libmad.html). 

I adapted it so that it creates the missing files for autoreconf.

The optimizations use old options for gcc that do not work with newer compilers or x86_64. See e.g. [this Debian bug report](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=415279). For example, Debian packagers have disabled these optimizations since 2008 (see package version [0.15.1b-3 from Debian changelog](https://metadata.ftp-master.debian.org/changelogs//main/libm/libmad/libmad_0.15.1b-10_changelog)).

Reply:
Can you add the URL where the patch comes from to the package?

Reply:
I added a comment on the patch origins to the package. The patch is not identical to the one from the web page as it also does the commands
```sh
sed "s@AM_CONFIG_HEADER@AC_CONFIG_HEADERS@g" -i configure.ac
touch NEWS AUTHORS ChangeLog
```
that are needed for the installation.